src/charm.py

```diff
@@ -575,29 +599,103 @@ def _on_mongod_pebble_ready(self, event) -> None:
 
         except (PathError, ProtocolError, MissingSecretError) as e:
             logger.error("Cannot initialize workload: %r", e)
-            event.defer()
-            return
-
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("mongod", self._mongod_layer, combine=True)
-        container.add_layer("log_rotate", self._log_rotate_layer, combine=True)
+            raise FailedToUpdateFilesystem
+
+    def __configure_layers(self, container: Container) -> None:
```

Comment:
I can rename all that.
@Mehdi-Bendriss Is there a specific reason to avoid "private" methods in charming?
Since those methods are only here for layering and reducing code complexity, it seemed a perfect use of private methods but if there's a rationale behind this, I'll gladly read it :)

Reply:
Done in 47317355